### ISSUE TITLE
Highlight connected node in signal connection list

### ISSUE THREAD

**Issue by dugramen** [godotengine/godot-proposals#10129]
### Describe the project you are working on

Editor Plurgins

### Describe the problem or limitation you are having in your project

In a node's signal connection list, it's not obvious which node a signal is connected to. You can double click a connection and it will open the script and method it is connected to. But there are also connections without any scripts, so double clicking does nothing. All you can get is a relative path to the node, which is hard to understand and is usually cut off

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When a connection is selected, highlight the node in the tree.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

A simple rectangle outline in the scene tree. Although I'm fine with any indicator
![Screenshot 2024-07-06 120946-2](https://github.com/godotengine/godot-proposals/assets/54819319/b529f482-9de0-4e9e-8b49-80f6192d849b)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so

### Is there a reason why this should be core and not an add-on in the asset library?

The Tree node is very difficult to work with, so I don't know if it's possible as an addon

**Comment by passivestar** [comment]
Perhaps the easiest way to implement this would be to flash a stylebox on top of the tree item (trees have `get_item_area_rect()`) for a second instead of tying into the tree drawing logic itself. This flashing logic can then be reused for other parts of the editor for consistent look, and the stylebox can be made themable

**Comment by dugramen** [comment]
I agree with flashing a stylebox for a second, and I found another place it might be needed. When `@export`ing a node to the inspector, there is a 3-dot menu with an option called "Show Node in Tree". Clicking it highlights the node in the tree as if you clicked the node normally, and they look very similar. If the node is already selected, there is no visual change

![Screenshot 2024-07-07 150002](https://github.com/godotengine/godot-proposals/assets/54819319/266e1e3b-f067-4780-a2be-c13c18c360ff)

Both cases should probably be highlighted the same way. The flash would be better in case you're already selecting that node, but if that's not ok then I guess it could just reuse this existing method
